src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -318,4 +321,40 @@ public void storeArticle(Article article) {
     }
     return articles;
   }
+
+  /**
+   * Stores a forum in the Datastore
+   *
+   * @param forum forum to be stored
+   */
+  public void storeForum(Forum forum) {
+    Entity forumEntity = new Entity("Forum", forum.getId().toString());
+    forumEntity.setProperty("title", forum.getTitle());
+    forumEntity.setProperty("owners", forum.getOwners());
+    forumEntity.setProperty("members", forum.getMembers());
+    forumEntity.setProperty("keywords", forum.getKeywords());
+    forumEntity.setProperty("articleIds", forum.getArticleIds());
+
+    datastore.put(forumEntity);
+  }
+
+  /**
+   * Retrieves a Forum by string id
+   *
+   * @param id id of a forum
+   * @return Forum object
+   * @throws EntityNotFoundException when no such forum with id is found
+   */
+  public Forum getForum(String id) throws EntityNotFoundException {
+    Key key = KeyFactory.createKey("Forum", id);
+    Entity forumEntity = datastore.get(key);
+    UUID uuid = UUID.fromString(id);
+    String title = (String) forumEntity.getProperty("title");
+    List<User> owners = (List<User>) forumEntity.getProperty("owners");
```

Comment:
Oh wait, can you do this?

Reply:
Not really used it as a placeholder of sorts but I changed it in the future commits.